### PR TITLE
send case format in the request for case creation

### PR DESCRIPTION
**Pull request by jamal-khey**

### CODE REVIEW

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -442,6 +444,7 @@ export function duplicateStudy(
     studyDescription,
     sourceStudyUuid,
     parentDirectoryUuid
+    //(jamal) add case format here also ?
```
Comment (flomillot) [inline src/utils/rest-api.js]: Forgot ?

Comment (jamal-khey) [reply src/utils/rest-api.js]: yes i fixed it

File: src/components/dialogs/create-study-dialog/create-study-dialog.js [hunk 2]
```diff
@@ -145,9 +146,11 @@ const CreateStudyDialog = ({ open, onClose, providedExistingCase }) => {
                     setValue(FORMATTED_CASE_PARAMETERS, formattedParams, {
                         shouldDirty: true,
                     });
+                    setCaseFormat(result.formatName);
```
Comment (anistouri) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js]: Why don't you add it with the other fields?
```suggestion
                    setValue(FORMAT_NAME, result.formatName);
```

**Review by anistouri — APPROVED** [review]

File: src/components/dialogs/create-study-dialog/create-study-dialog.js (hunk 2 above)
Comment (jamal-khey) [reply src/components/dialogs/create-study-dialog/create-study-dialog.js]: done, i removed the state

File: src/components/dialogs/create-study-dialog/create-study-dialog-utils.js [hunk 3]
```diff
@@ -34,4 +36,5 @@ export const createStudyDialogFormValidationSchema = yup.object().shape({
     [CASE_UUID]: yup.string().required(),
     [CASE_FILE]: yup.mixed().nullable().required(),
     [DIRECTORY]: yup.string().required(),
+    [CASE_FORMAT]: yup.string().optional(),
```
Comment (flomillot) [inline src/components/dialogs/create-study-dialog/create-study-dialog-utils.js]: I don't think we use optional on string in our projects, why adding it as we use empty string ?
https://github.com/jquense/yup#schemaoptional-schema

Comment (anistouri) [reply src/components/dialogs/create-study-dialog/create-study-dialog-utils.js]: Why do you want the field to be optional? Are there cases where we can be without the format? I believe it should be required

Comment (jamal-khey) [reply src/components/dialogs/create-study-dialog/create-study-dialog-utils.js]: i changed it to be required

**Review by flomillot — APPROVED** [review]
Code OK

**Review by anistouri — APPROVED** [review]

**Review by Meklo — APPROVED** [review]

**Comment by Meklo** [comment]
I may have approved a bit hastily, I noticed afterward that setting CASE_FORMAT to required has produced a side effect.
Now when importing a valid case the form doesn't automatically enable the Validate button, do you get that too ?

**Comment by jamal-khey** [comment]
> I may have approved a bit hastily, I noticed afterward that setting CASE_FORMAT to required has produced a side effect. Now when importing a valid case the form doesn't automatically enable the Validate button, do you get that too ?

I wasn't aware of this side effect. but i think it does not affect the user experience that much, the button become quickly available

**Pull request merged**